resources/com/github/coreos/buildconfig.json

```diff
@@ -0,0 +1,163 @@
+{
+  "kind": "Template",
+  "apiVersion": "v1",
+  "metadata": {
+    "name": "${NAME}",
+    "annotations": {
+      "openshift.io/display-name": "Buildconfig for ${NAME}",
+      "description": "${NAME}",
+      "template.openshift.io/bindable": "false"
+    }
+  },
+  "labels": {
+      "template": "${NAME}",
+      "app": "${NAME}"
+  },
+  "objects": [
+    {
+      "kind": "ImageStream",
+      "apiVersion": "v1",
+      "metadata": {
+        "name": "${NAME}",
+        "annotations": {
+          "description": "Keeps track of changes in the application image"
+        },
+	"ownerReferences": [
+          {
+           "apiVersion": "batch/v1",
+           "kind": "Job",
+           "name": "${OWNERNAME}",
+           "uid": "${UID}",
+           "controller": true
+          }
+        ]
+      },
+      "spec": {
+        "failedBuildHistoryLimit": 1,
+        "successfulBuildsHistoryLimit": 1,
+        "lookupPolicy": {
+          "local": true
+        }
+      }
+    },
+    {
+      "kind": "BuildConfig",
+      "apiVersion": "v1",
+      "metadata": {
+        "name": "${NAME}",
+        "annotations": {
+          "description": "Defines how to build the application",
+          "template.alpha.openshift.io/wait-for-ready": "true"
+        },
+        "ownerReferences": [
+          {
+           "apiVersion": "batch/v1",
+           "kind": "Job",
+           "name": "${OWNERNAME}",
+           "uid": "${UID}",
+           "controller": true
+          }
+        ]
+      },
+      "spec": {
+        "source": {
+          "type": "Git",
+          "git": {
+            "uri": "${SOURCE_REPOSITORY_URL}",
+            "ref": "${SOURCE_REPOSITORY_REF}"
+          }
+        },
```

Comment:
Hmm actually, we should be able to just drop this and instead do `openshift.startBuild("--from-dir=.")` (could make the dir path overridable via a parameter to `buildImage()`) which will pass the workspace as the build input. The major advantage of this is that it closes a race condition where the ref was moved since the job was triggered. It's also simpler because we don't have to fetch around for env vars and parse it out; basically it puts the onus on the caller to fetch the source. :)

Reply:
Thanks again @jlebon \o/,  awesome,  it also solves the issue to get the PR reference. It is much cleaner in this way. 
The only think missing is the stuff related to[ #61](https://github.com/coreos/coreos-ci-lib/pull/61#discussion_r569401940).  I will get some help from the community to understand it.